src/System.Text.Json/src/System/Text/Json/Node/JsonObject.cs

```diff
@@ -0,0 +1,465 @@
+        /// </exception>
+        /// <exception cref="ArgumentNullException">
+        ///   Some of property names are null.
+        /// </exception>
+        public void AddRange(IEnumerable<KeyValuePair<string, JsonNode>> jsonProperties)
+        {
+            foreach (KeyValuePair<string, JsonNode> property in jsonProperties)
+            {
+                Add(property);
+            }
+        }
+
+        /// <summary>
+        ///   Removes the property with the specified name.
+        /// </summary>
+        /// <param name="propertyName"></param>
+        /// <returns>
+        ///   <see langword="true"/> if the property is successfully found in a JSON object and removed,
+        ///   <see langword="false"/> otherwise.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        ///   Provided property name is null.
+        /// </exception>
+        public bool Remove(string propertyName) => propertyName != null ? _dictionary.Remove(propertyName) : throw new ArgumentNullException(nameof(propertyName));
+
+        /// <summary>
+        ///   Determines whether a property is in a JSON object.
+        /// </summary>
+        /// <param name="propertyName">Name of the property to check.</param>
+        /// <returns>
+        ///   <see langword="true"/> if the property is successfully found in a JSON object,
+        ///   <see langword="false"/> otherwise.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        ///   Provided property name is null.
+        /// </exception>
+        public bool ContainsProperty(string propertyName) => propertyName != null ? _dictionary.ContainsKey(propertyName) : throw new ArgumentNullException(nameof(propertyName));
+
+        /// <summary>
+        ///   Modifies the name of specified property.
+        /// </summary>
+        /// <param name="oldName">Old name of the property to change.</param>
+        /// <param name="newName">New name of the property to change.</param>
+        /// <exception cref="ArgumentException">
+        ///   Property name to set already exists if handling duplicates is set to <see cref="DuplicatePropertyNameHandling.Error"/>.
+        /// </exception>
+        /// <exception cref="KeyNotFoundException">
+        ///   Property with specified name is not found in JSON object.
+        /// </exception>
+        /// <remnark>
+        ///   Does not guarantee keeping the order the same.
+        /// </remnark>
+        public void ModifyPropertyName(string oldName, string newName)
+        {
+            if (!_dictionary.ContainsKey(oldName))
+                throw new KeyNotFoundException(string.Format(SR.PropertyNotFound, oldName));
+
+            JsonNode previousValue = _dictionary[oldName];
+            _dictionary.Remove(oldName);
+            _dictionary.Add(newName, previousValue);
```

Comment:
Is this the right implementation?  What should it do when the duplicate mode is replace?

Maybe just delete this method and let the caller do it themselves.

Reply:
We thought it might be useful to have an accelerating method for changing property name. I think it should just change the name in all of the handling duplicates cases (because we are not adding a duplicate, but just renaming existing property).

Reply:
But if the new property name is already existing, it should behave exactly as `Add`.

Reply:
> But if the new property name is already existing, it should behave exactly as Add.

So it should call Add instead of _dictionary.Add.

```C#
JsonObject obj = new JsonObject(Duplicates.Replace);
obj["prop1"] = value1;
obj["prop2"] = value2;
obj.RenameProperty("prop1", "prop2");
```

Right now this throws.

Given that we've gone 15 years without adding similar functionality to `Dictionary<TKey, TValue>` I don't think there's a real market for it.

Reply:
> So it should call Add instead of _dictionary.Add.

Oh, yes.

> Given that we've gone 15 years without adding similar functionality to Dictionary<TKey, TValue> I don't think there's a real market for it.

:disappointed:

Reply:
But isn't changing property name (in the API that's aimed for being modifiable) more likely to be needed and used that in a regular dictionary?

Reply:
> But isn't changing property name in JSON object (in the API that's aimed for being modifiable) more likely to be needed and used that in a regular dictionary?

It seems to be the same to me (since the dictionary type is itself designed to be mutable :smile:).  Maybe in the JSON object domain there's more of a need than in the fully general/generic case; but since it's only saving three lines of code I'd wait until someone actually asks for it.